mongo/tests/test_integration.py

```diff
@@ -616,9 +616,8 @@ def test_mongod_bad_auth(check, dd_run_check, aggregator, username, password):
         'password': password,
         'options': {'authSource': 'authDB'},
     }
-    with pytest.raises(Exception, match="Authentication failed"):
-        mongo_check = check(instance)
-        dd_run_check(mongo_check)
+    mongo_check = check(instance)
+    dd_run_check(mongo_check)
```

Comment:
We could check the service check is set to CRITICAL in this case.

Not sure if it's possible here, but we could also check the log to be sure that the error is an authentication error (just like we did previously using the exception), but you already covered this case in your unit tests

Reply:
https://github.com/DataDog/integrations-core/pull/12767/commits/7f0ac230bfcb62572a7021189861e685df266ae6